packages/taquito-sapling/src/taquito-sapling.ts

```diff
@@ -3,6 +3,323 @@
+      balance,
+      root,
+      boundData,
+      signature,
+    });
+
+    return forgedSaplingTx.toString('hex');
+  }
+
+  /**
+   * @description Prepare a sapling transaction (zet to zet)
+   * @param saplingTxParams `to` is the payment address that will receive the shielded tokens (zet).
+   * `amount` is the amount of unshielded tokens in tez by default.
+   * `mutez` needs to be set to true if the amount of unshielded tokens is in mutez.
+   * `memo` is an empty string by default.
+   * @returns a string representing the sapling transaction.
+   */
+  async prepareSaplingTransaction(saplingTxParams: ParametersSaplingTransaction[]) {
+    const { formatedParams, totalAmount } = this.formatTransactionParams(
+      saplingTxParams,
+      this.validateDestinationSaplingAddress
+    );
+
+    const root = await this.getRoot();
+    const chosenInputs = await this.selectInputsToSpend(totalAmount);
+    const { inputs, outputs, signature, balance } = await this.#saplingTxBuilder.createSaplingTx(
+      formatedParams,
+      totalAmount,
+      defaultBoundData,
+      chosenInputs
+    );
+
+    const forgedSaplingTx = this.#saplingForger.forgeSaplingTransaction({
+      inputs,
+      outputs,
+      balance,
+      root,
+      boundData: defaultBoundData,
+      signature,
+    });
+
+    return forgedSaplingTx.toString('hex');
+  }
+
+  private formatTransactionParams(
+    txParams: ParametersSaplingTransaction[],
+    validateDestination: (to: string) => void
+  ) {
+    const formatedParams: {
+      to: string;
+      amount: string;
+      memo: string;
+    }[] = [];
+
+    let totalAmount = new BigNumber(0);
+
+    txParams.forEach((param) => {
+      validateDestination(param.to);
+      const amountMutez = param.mutez
+        ? param.amount.toString()
+        : format('tz', 'mutez', param.amount).toString();
+      totalAmount = totalAmount.plus(new BigNumber(amountMutez));
+      const memo = param.memo ?? defaultMemo;
+      if (memo.length > this.#memoSize) {
+        throw new InvalidMemo(memo, 'The memo is too long.');
+      }
+
+      formatedParams.push({ to: param.to, amount: amountMutez, memo });
+    });
+
+    return { formatedParams, totalAmount };
+  }
+
+  private async getRoot() {
+    if (this.#saplingId) {
+      const { root } = await this.#readProvider.getSaplingDiffById({ id: this.#saplingId }, 'head');
+      return root;
+    } else {
+      const { root } = await this.#readProvider.getSaplingDiffByContract(
+        this.#contractAddress,
+        'head'
+      );
+      return root;
+    }
+  }
+  private async createBoundData(destination: string) {
+    const pref = destination.substr(0, 3);
+
+    let pad: Buffer;
+    switch (pref) {
+      case 'tz1': {
+        pad = Buffer.from('00', 'hex');
+        break;
+      }
+      case 'tz2': {
+        pad = Buffer.from('01', 'hex');
+        break;
+      }
+      case 'tz3': {
+        pad = Buffer.from('02', 'hex');
+        break;
+      }
```

Comment:
since this is used in a couple places could this be turned into a helper function and included in all applicable places?